main.c

```diff
@@ -134,6 +142,21 @@ main(int argc, char *argv[])
 	if (osUpdateScreenInit())
 		return -1;
 
+	/* Allow SIGTERM to interrupt select() and move to cleanup */
+	sigemptyset(&mask);
+	sigaddset(&mask, SIGTERM);
+	if (sigprocmask(SIG_BLOCK, &mask, NULL) == -1) {
```

Comment:
I wonder if I should do this only when select() is used... It might prevent usleep() from being interrupted by SIGTERM.

Reply:
Or then use timed select everywhere, instead of usleep.

Reply:
(or I think you could use a "classic" signal handler and call osUpdateScreenExit() and exit() there)

Reply:
> Or then use timed select everywhere, instead of usleep.
Should be doable and I like the idea of unifying these cases.

> (or I think you could use a "classic" signal handler and call osUpdateScreenExit() and exit() there)
Yeah, that also works and I did have version like that already, but I thought it would be nicer to use the cleanup in main(). That ought to at least avoid problems from doing something possibly forbidden in signal handler.

Reply:
> That ought to at least avoid problems from doing something possibly forbidden in signal handler.

Yeah, at least more future proof to avoid that, even if it would work for now.

Reply:
Converted select() and usleep() calls to pselect(), added close(sigfd), moved sigprocmask() after signalfd() call and added handling for SIGINT too.